apps/dash-image-segmentation/trainable_segmentation.py

```diff
@@ -0,0 +1,289 @@
+from itertools import combinations_with_replacement
```

Comment:
this is the module I submitted to a scikit-image PR, it's more optimized than the previous version and it has a more flexible API.